### DISCUSSION TITLE
Containerd tasks metrics using grpcurl command

### DISCUSSION THREAD

**Discussion by Sicarius07** [aquasecurity/trivy#5436] [Q&A]
### Question

I want to get the METRIC data of the currently running tasks having similar metrics as given below-
```
METRIC                VALUE
pids.current          22
pids.limit            18446744073709551615
cpu.usage_usec        5759406
cpu.user_usec         5424353
cpu.system_usec       335053
cpu.nr_periods        0
cpu.nr_throttled      0
cpu.throttled_usec    0
memory.usage          192995328
memory.usage_limit    18446744073709551615
memory.swap_usage     0
memory.swap_limit     18446744073709551615
```

This command gives the metrics of currenty running tasks-
`sudo go/bin/grpcurl  --plaintext -import-path "/home/ubuntu/containerd/api/services/tasks/v1" --proto /home/ubuntu/containerd/api/services/tasks/v1/tasks.proto -rpc-header 'containerd-namespace: default' --unix /run/containerd/containerd.sock containerd.services.tasks.v1.Tasks/Metrics`

but metrics type "io.containerd.cgroups.v2.Metrics" is not recognized/supported.

This is the output of the above command-

```
  "metrics": [
    {
      "timestamp": "2023-10-10T06:56:38.245697786Z",
      "id": "MA_test_hello2",
      "data": {
        "@error": "io.containerd.cgroups.v2.Metrics is not recognized; see @value for raw binary message data",
        "@type": "io.containerd.cgroups.v2.Metrics",
        "@value": "Cg0IFhD///////////8BEg4I/f7QAhDCsLwCGLrOFCJiCICAxDwQgMC5IxiAgBYg4LRwOIDg4ApAgOABWICgwzxggGBogIDXHXCAwOIFgAH4p1SIAeiMHJABjbICmAHMAdAB2BKAAoCAqmGIAv.........."
      }
    }
  ]
}
```

Can you please help me in resolving the same?

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

**Comment by nikpivkin** [comment]
@Sicarius07 I think you should open it here https://github.com/containerd/containerd
